File: code/aiming_transporting/matlab/modules/saving_data.m
function write_to_csv(params, theta, phi, tau_us, r_eq_cm, R_T_cm, fluence_rate, filename)
% 保存仿真结果到CSV文件

% 创建结果表格
result_table = table(...
    params.E, params.B, params.R, params.Psi, ...
    params.n_i, params.Ib, params.em_type, params.R0, ...
    theta, phi, tau_us, r_eq_cm, R_T_cm, fluence_rate, ...
    'VariableNames', {...
    'E_MeV', 'B_g', 'R_km', 'Psi_rad', ...
    'n_i_cm3', 'Ib_A', 'em_type', 'R0_cm', ...
    'theta_rad', 'phi_rad', 'tau_us', 'r_eq_cm', 'R_T_cm', 'fluence_rate'});

% 写入文件
writetable(result_table, filename);

% 添加单位说明
fid = fopen(filename, 'a');
fprintf(fid, '\n\nUnits:\n');
fprintf(fid, 'E: MeV\nB: gauss\nR: km\nPsi: rad\n');
fprintf(fid, 'n_i: cm^-3\nIb: A\nR0: cm\n');
fprintf(fid, 'theta: rad\nphi: rad\ntau: μs\n');
fprintf(fid, 'r_eq: cm\nR_T: cm\nfluence_rate: cm^-2s^-1');
fclose(fid);
end

File: code/aiming_transporting/modules/saving_data.py
from modules.beam_params import BeamParams
import numpy as np
import os
import csv

def write_to_csv(params: BeamParams, theta, phi, tau, r_eq, R_T, fluence_rate, file_path='results.csv'):
    header = [
        "电子束能量(MeV)", "磁场(g)", "目标距离(km)",
        "目标角度(rad)", "极角(deg)", "方位角(deg)", "打击时间(μs)","初始束流半径(cm)","平衡束流半径(cm)",
        "背景离子密度(cm^-3)", "电子束电流(A)", "发射度类型",
        "到靶束流包络半径(cm)", "到靶注量率(cm^-2s^-1)"
    ]
    data = [
        params.E, params.B, params.R, params.Psi,
        round(np.degrees(theta), 1) if theta is not None else 'nan',
        round(np.rad2deg(np.pi/2-phi), 1) if phi is not None else 'nan',
        round(tau, 1) if tau is not None else 'nan',
        params.R0, round(r_eq,1) if r_eq is not None else 'nan',
        f'{params.n_i:.1e}' if params.n_i is not None else 'nan',
        params.Ib, params.em_type,
        round(R_T, 1) if R_T is not None else 'nan',
        f'{fluence_rate:.1e}' if fluence_rate is not None else 'nan'
    ]
    with open(file_path, 'a', newline='', encoding='utf_8_sig') as f:
        writer = csv.writer(f)
        if not os.path.exists(file_path) or os.path.getsize(file_path) == 0:
            writer.writerow(header)
        writer.writerow(data)